Missing tf_python_protos_cc library dependency in tf_tutorials.cmake

### System information
Windows10
VisualStudio 2017
TensorFlow 1.3.0
Python 3.5.3
CMake 3.8.1

### Describe the problem
I can't build **tf_tutorials_example_trainer** due missing library dependency to tf_python_protos_cc.lib. A lot of linking errors occurred. If the bold line is added into tf_tutorials.cmake then the compilation works:

target_link_libraries(tf_tutorials_example_trainer PUBLIC
    tf_protos_cc
    **tf_python_protos_cc**
    ${tf_core_gpu_kernels_lib}
    ${tensorflow_EXTERNAL_LIBRARIES}
)

CMake is used build the project files for VisualStudio. 

![image](https://user-images.githubusercontent.com/30691148/28971246-95f12174-792b-11e7-8d9f-5bb5542a5ce0.png)

### Source code / logs
`1>------ Build started: Project: tf_tutorials_example_trainer, Configuration: Release x64 ------
1>   Creating library C:/Development/dev/test_projects/deeplearning/tensorflow_build64/Release/tf_tutorials_example_trainer.lib and object C:/Development/dev/test_projects/deeplearning/tensorflow_build64/Release/tf_tutorials_example_trainer.exp
1>ensemble_optimizer_ops.cc.obj : error LNK2019: unresolved external symbol "public: __cdecl tensorflow::boosted_trees::trees::DecisionTreeConfig::DecisionTreeConfig(void)" (??0DecisionTreeConfig@trees@boosted_trees@tensorflow@@QEAA@XZ) referenced in function "public: static class tensorflow::boosted_trees::trees::DecisionTreeConfig * __cdecl google::protobuf::Arena::CreateMessage<class tensorflow::boosted_trees::trees::DecisionTreeConfig>(class google::protobuf::Arena *)" (??$CreateMessage@VDecisionTreeConfig@trees@boosted_trees@tensorflow@@@Arena@protobuf@google@@SAPEAVDecisionTreeConfig@trees@boosted_trees@tensorflow@@PEAV012@@Z)
1>prediction_ops.cc.obj : error LNK2001: unresolved external symbol "public: __cdecl tensorflow::boosted_trees::trees::DecisionTreeConfig::DecisionTreeConfig(void)" (??0DecisionTreeConfig@trees@boosted_trees@tensorflow@@QEAA@XZ)
1>training_ops.cc.obj : error LNK2001: unresolved external symbol "public: __cdecl tensorflow::boosted_trees::trees::DecisionTreeConfig::DecisionTreeConfig(void)" (??0DecisionTreeConfig@trees@boosted_trees@tensorflow@@QEAA@XZ)
1>ensemble_optimizer_ops.cc.obj : error LNK2019: unresolved external symbol "public: void __cdecl tensorflow::boosted_trees::trees::DecisionTreeConfig::Swap(class tensorflow::boosted_trees::trees::DecisionTreeConfig *)" (?Swap@DecisionTreeConfig@trees@boosted_trees@tensorflow@@QEAAXPEAV1234@@Z) referenced in function "public: virtual void __cdecl tensorflow::AddTreesToEnsembleOp::Compute(class tensorflow::OpKernelContext * const)" (?Compute@AddTreesToEnsembleOp@tensorflow@@UEAAXQEAVOpKernelContext@2@@Z)
1>ensemble_optimizer_ops.cc.obj : error LNK2019: unresolved external symbol "protected: __cdecl tensorflow::boosted_trees::trees::DecisionTreeConfig::DecisionTreeConfig(class google::protobuf::Arena *)" (??0DecisionTreeConfig@trees@boosted_trees@tensorflow@@IEAA@PEAVArena@protobuf@google@@@Z) referenced in function "public: static class tensorflow::boosted_trees::trees::DecisionTreeConfig * __cdecl google::protobuf::Arena::CreateMessage<class tensorflow::boosted_trees::trees::DecisionTreeConfig>(class google::protobuf::Arena *)" (??$CreateMessage@VDecisionTreeConfig@trees@boosted_trees@tensorflow@@@Arena@protobuf@google@@SAPEAVDecisionTreeConfig@trees@boosted_trees@tensorflow@@PEAV012@@Z)
1>training_ops.cc.obj : error LNK2001: unresolved external symbol "protected: __cdecl tensorflow::boosted_trees::trees::DecisionTreeConfig::DecisionTreeConfig(class google::protobuf::Arena *)" (??0DecisionTreeConfig@trees@boosted_trees@tensorflow@@IEAA@PEAVArena@protobuf@google@@@Z)
1>ensemble_optimizer_ops.cc.obj : error LNK2019: unresolved external symbol "public: __cdecl tensorflow::boosted_trees::trees::DecisionTreeMetadata::DecisionTreeMetadata(void)" (??0DecisionTreeMetadata@trees@boosted_trees@tensorflow@@QEAA@XZ) referenced in function "public: static class tensorflow::boosted_trees::trees::DecisionTreeMetadata * __cdecl google::protobuf::Arena::CreateMessage<class tensorflow::boosted_trees::trees::DecisionTreeMetadata>(class google::protobuf::Arena *)" (??$CreateMessage@VDecisionTreeMetadata@trees@boosted_trees@tensorflow@@@Arena@protobuf@google@@SAPEAVDecisionTreeMetadata@trees@boosted_trees@tensorflow@@PEAV012@@Z)
1>training_ops.cc.obj : error LNK2001: unresolved external symbol "public: __cdecl tensorflow::boosted_trees::trees::DecisionTreeMetadata::DecisionTreeMetadata(void)" (??0DecisionTreeMetadata@trees@boosted_trees@tensorflow@@QEAA@XZ)
1>ensemble_optimizer_ops.cc.obj : error LNK2019: unresolved external symbol "protected: __cdecl tensorflow::boosted_trees::trees::DecisionTreeMetadata::DecisionTreeMetadata(class google::protobuf::Arena *)" (??0DecisionTreeMetadata@trees@boosted_trees@tensorflow@@IEAA@PEAVArena@protobuf@google@@@Z) referenced in function "public: static class tensorflow::boosted_trees::trees::DecisionTreeMetadata * __cdecl google::protobuf::Arena::CreateMessage<class tensorflow::boosted_trees::trees::DecisionTreeMetadata>(class google::protobuf::Arena *)" (??$CreateMessage@VDecisionTreeMetadata@trees@boosted_trees@tensorflow@@@Arena@protobuf@google@@SAPEAVDecisionTreeMetadata@trees@boosted_trees@tensorflow@@PEAV012@@Z)
1>training_ops.cc.obj : error LNK2001: unresolved external symbol "protected: __cdecl tensorflow::boosted_trees::trees::DecisionTreeMetadata::DecisionTreeMetadata(class google::protobuf::Arena *)" (??0DecisionTreeMetadata@trees@boosted_trees@tensorflow@@IEAA@PEAVArena@protobuf@google@@@Z)

**...**

1>C:\Development\dev\test_projects\deeplearning\tensorflow_build64\Release\tf_tutorials_example_trainer.exe : fatal error LNK1120: 85 unresolved externals
1>Done building project "tf_tutorials_example_trainer.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
`

--- a/tensorflow/contrib/cmake/tf_core_framework.cmake
+++ b/tensorflow/contrib/cmake/tf_core_framework.cmake
@@ -87,6 +87,7 @@ endfunction()
 
 file(GLOB_RECURSE tf_protos_cc_srcs RELATIVE ${tensorflow_source_dir}
     "${tensorflow_source_dir}/tensorflow/core/*.proto"
+    "${tensorflow_source_dir}/tensorflow/contrib/boosted_trees/proto/*.proto"
 )
 RELATIVE_PROTOBUF_GENERATE_CPP(PROTO_SRCS PROTO_HDRS
     ${tensorflow_source_dir} ${tf_protos_cc_srcs}

--- a/tensorflow/contrib/cmake/tf_core_kernels.cmake
+++ b/tensorflow/contrib/cmake/tf_core_kernels.cmake
@@ -166,9 +166,10 @@ if(WIN32)
 endif(WIN32)
 
 file(GLOB_RECURSE tf_core_gpu_kernels_srcs
-   "${tensorflow_source_dir}/tensorflow/core/kernels/*.cu.cc"
-   "${tensorflow_source_dir}/tensorflow/contrib/rnn/kernels/*.cu.cc"
-   "${tensorflow_source_dir}/tensorflow/contrib/seq2seq/kernels/*.cu.cc"
+    "${tensorflow_source_dir}/tensorflow/core/kernels/*.cu.cc"
+    "${tensorflow_source_dir}/tensorflow/contrib/framework/kernels/zero_initializer_op_gpu.cu.cc"
+    "${tensorflow_source_dir}/tensorflow/contrib/rnn/kernels/*.cu.cc"
+    "${tensorflow_source_dir}/tensorflow/contrib/seq2seq/kernels/*.cu.cc"
 )
 
 if(WIN32 AND tensorflow_ENABLE_GPU)

--- a/tensorflow/contrib/cmake/tf_core_kernels.cmake
+++ b/tensorflow/contrib/cmake/tf_core_kernels.cmake
@@ -135,6 +135,7 @@ file(GLOB_RECURSE tf_core_kernels_exclude_srcs
    "${tensorflow_source_dir}/tensorflow/core/kernels/*test_utils.cc"
    "${tensorflow_source_dir}/tensorflow/core/kernels/*main.cc"
    "${tensorflow_source_dir}/tensorflow/core/kernels/*.cu.cc"
+   "${tensorflow_source_dir}/tensorflow/core/kernels/fuzzing/*"
    "${tensorflow_source_dir}/tensorflow/core/kernels/hexagon/*"
    "${tensorflow_source_dir}/tensorflow/core/kernels/remote_fused_graph_rewriter_transform*.cc"
 )

--- a/tensorflow/contrib/cmake/tf_python.cmake
+++ b/tensorflow/contrib/cmake/tf_python.cmake
@@ -142,7 +142,6 @@ RELATIVE_PROTOBUF_GENERATE_PYTHON(
 file(GLOB_RECURSE tf_python_protos_cc_srcs RELATIVE ${tensorflow_source_dir}
     "${tensorflow_source_dir}/tensorflow/core/profiler/*.proto"
     "${tensorflow_source_dir}/tensorflow/python/*.proto"
-    "${tensorflow_source_dir}/tensorflow/contrib/boosted_trees/proto/*.proto"
     "${tensorflow_source_dir}/tensorflow/contrib/session_bundle/*.proto"
     "${tensorflow_source_dir}/tensorflow/contrib/tensorboard/*.proto"
     "${tensorflow_source_dir}/tensorflow/contrib/training/*.proto"

